### PR TITLE
Expose additional driver results for some methods

### LINKED ISSUE #12: Support for other WriteResult parameters
Write operations like update, updateOne, replaceOne, etc. often have more results than `modifiedCount`. In my case, I'd like to access `matchedCount` to see if my filter actually got any hits. What do you think about returning to these callbacks not only `modifiedCount`, but as a third parameter the raw operation result so we can access these? The callback would use the signature `function (error, count, results)` in these cases.

I'm able to open a PR with updated testing and documentation if you think that's a good idea.

### PR DESCRIPTION
**Pull request by henriquesa**
Closes #12

### CODE REVIEW

**Comment by jedireza** [comment]
Apologies for the wait. I'll try to test this out tonight.

**Comment by henriquesa** [comment]
Take your time! No hurry

File: API.md [hunk 1]
```diff
@@ -375,11 +375,16 @@ Replaces a document and returns the count of modified documents where:
  - `options` - an options object passed to MongoDB's native
    [`replaceOne`](https://docs.mongodb.com/manual/reference/method/db.collection.replaceOne/)
    method.
- - `callback` - the callback method using the signature `function (err, count)`
+ - `callback` - the callback method using the signature `function (err, count, result)`
    where:
     - `err` - if the query failed, the error reason, otherwise `null`.
     - `count` - if the query succeeded, a number indicating how many documents
       were modified.
+    - `result` - if the query succeeded, the raw result document returned
```
Comment (jedireza) [inline API.md]: Let's simplify the description for these to just:

`if the query succeeded, the raw result document returned by MongoDB's native driver.`

This way we're not responsible to document what's not in our control (if they add/remove things from the results document for example).

File: API.md [hunk 2]
```diff
@@ -406,11 +411,16 @@ Updates multiple documents and returns the count of modified documents where:
  - `options` - an options object passed to MongoDB's native
    [`updateMany`](https://docs.mongodb.com/manual/reference/method/db.collection.updateMany/)
    method.
- - `callback` - the callback method using the signature `function (err, count)`
+ - `callback` - the callback method using the signature `function (err, count, result)`
    where:
     - `err` - if the query failed, the error reason, otherwise `null`.
     - `count` - if the command succeeded, a number indicating how many documents
       were modified.
+    - `result` - if the query succeeded, the raw result document returned
```
Comment (jedireza) [inline API.md]: Let's simplify the description (same as above).

File: API.md [hunk 3]
```diff
@@ -421,11 +431,16 @@ Updates a document and returns the count of modified documents where:
  - `options` - an options object passed to MongoDB's native
    [`updateOne`](https://docs.mongodb.com/manual/reference/method/db.collection.updateOne/)
    method.
- - `callback` - the callback method using the signature `function (err, count)`
+ - `callback` - the callback method using the signature `function (err, count, result)`
    where:
     - `err` - if the query failed, the error reason, otherwise `null`.
     - `count` - if the command succeeded, a number indicating how many documents
       were modified.
+    - `result` - if the query succeeded, the raw result document returned
```
Comment (jedireza) [inline API.md]: Let's simplify the description (same as above).

File: test/index.js [hunk 4]
```diff
@@ -564,11 +564,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
                 $set: { isCool: true }
             };
 
-            SubModel.updateOne(filter, update, (err, count) => {
+            SubModel.updateOne(filter, update, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(1);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm `result` is an object and not that it includes specific properties (again incase they change the document in the future).

File: test/index.js [hunk 5]
```diff
@@ -603,11 +608,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
             };
             const options = { upsert: true };
 
-            SubModel.updateOne(filter, update, options, (err, count) => {
+            SubModel.updateOne(filter, update, options, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(1);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm result is an object (same as above).

File: test/index.js [hunk 6]
```diff
@@ -678,11 +689,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
             const filter = {};
             const update = { $set: { isCool: true } };
 
-            SubModel.updateMany(filter, update, (err, count) => {
+            SubModel.updateMany(filter, update, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(3);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm result is an object (same as above).

File: test/index.js [hunk 7]
```diff
@@ -713,11 +729,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
             const update = { $set: { isCool: true } };
             const options = { upsert: true };
 
-            SubModel.updateMany(filter, update, options, (err, count) => {
+            SubModel.updateMany(filter, update, options, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(3);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm result is an object (same as above).

File: test/index.js [hunk 8]
```diff
@@ -1177,11 +1199,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
             const filter = { name: 'Ren' };
             const doc = { isCool: true };
 
-            SubModel.replaceOne(filter, doc, (err, count) => {
+            SubModel.replaceOne(filter, doc, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(1);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm result is an object (same as above).

File: test/index.js [hunk 9]
```diff
@@ -1208,11 +1235,16 @@ lab.experiment('MongoModels Proxied Methods', () => {
             const doc = { isCool: true };
             const options = { upsert: true };
 
-            SubModel.replaceOne(filter, doc, options, (err, count) => {
+            SubModel.replaceOne(filter, doc, options, (err, count, result) => {
 
                 Code.expect(err).to.not.exist();
                 Code.expect(count).to.be.a.number();
                 Code.expect(count).to.equal(1);
+                Code.expect(result).to.be.an.object();
```
Comment (jedireza) [inline test/index.js]: Let's only confirm result is an object (same as above).

**Pull request merged**